src/vi/truyentranh3q/src/eu/kanade/tachiyomi/extension/vi/truyentranh3q/TruyenTranh3Q.kt

```diff
@@ -0,0 +1,273 @@
+package eu.kanade.tachiyomi.extension.vi.truyentranh3q
+
+import android.net.ParseException
+import android.net.Uri
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.net.URLDecoder
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+class TruyenTranh3Q : ParsedHttpSource() {
+    override val name: String = "TruyenTranh3Q"
+    override val lang: String = "vi"
+    override val baseUrl: String = "https://truyentranh3q.com"
+    override val supportsLatest: Boolean = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder {
+        return super.headersBuilder().add("Referer", "$baseUrl/")
+    }
+
+    private val dateFormat = SimpleDateFormat("dd/MM/yyyy", Locale.US)
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/danh-sach/truyen-yeu-thich?page=$page", headers)
+    }
+
+    override fun popularMangaSelector(): String = "ul.list_grid.grid > li"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            element.select("h3 a").let {
+                title = it.text()
+                setUrlWithoutDomain(it.attr("abs:href"))
+            }
+            val relativeImageUrl = element.selectFirst(".book_avatar a img")?.attr("src")
+            thumbnail_url = relativeImageUrl?.let {
+                if (it.startsWith("/_next/image?")) {
+                    Uri.parse(it).getQueryParameter("url")?.let { encodedUrl ->
+                        URLDecoder.decode(encodedUrl, "UTF-8")
+                    }
+                } else {
+                    it
+                }
+            }
```

Comment:
Maybe something like this? Not sure if you need the URLDecoder part when using OkHttp's `HttpUrl`. Linter might disagree.
```kt
thumbnail_url = element.selectFirst(".book_avatar a img")
    ?.absUrl("src")
    ?.let { url ->
        url.toHttpUrlOrNull()
        ?.queryParameter("url")
        ?.let { URLDecoder.decode(it, "UTF-8") }
        ?: url
    }
```